Updated values in dbt_run_on_merge.yml (#7316)
fishtown-analytics/docs.getdbt.com: website/docs/guides/custom-cicd-pipelines.md

> It’s worth noting that we changed the `on:` section to now run **only** when there are pushes to a branch named `main` (i.e. a PR is merged). Have a look through [GitHub’s docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) on these filters for additional use cases.

Reviewer: I know this is not part of your PR but I can't unsee it!
```suggestion
It’s worth noting that we changed the `on:` section to now run **only** when there are pushes to a branch named `main` (for example, a pull request is merged). Have a look through [GitHub documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) on these filters for additional use cases.
```

> For the value of `DBT_JOB_BRANCH`, refer to the [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs) for information about `github` context property names and their use cases. 

Reviewer: ```suggestion
For information about `github` context property names and their use cases, refer to the [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs). 
```

> For the value of `DBT_JOB_BRANCH`, refer to the [GitHub documentation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs) for information about `github` context property names and their use cases. 

Reviewer: > For the value of `DBT_JOB_BRANCH`

I don't think they need to add a value for  `${{ github.ref_name }}` it should be generated automatically because it triggers the workflow. We need to update it to `${{ github.head_ref }}` which I added below!

      DBT_PROJECT_ID: 00000 # enter your project id
      DBT_PR_JOB_ID:  00000 # enter your job id
      DBT_API_KEY: ${{ secrets.DBT_API_KEY }}
      DBT_URL: https://cloud.getdbt.com # enter a URL that matches your job
      DBT_JOB_CAUSE: 'GitHub Pipeline CI Job' 

>       DBT_JOB_BRANCH: ${{ github.ref_name }}

Reviewer: ```suggestion
      DBT_JOB_BRANCH: ${{ github.head_ref }} # Resolves to the head_ref or source branch of the pull request in a workflow run. 
```
